cocotb/share/def/modelsim.def

```diff
@@ -1,115 +0,0 @@
-LIBRARY "mtipli.dll"
-EXPORTS
-vpi_chk_error
-vpi_compare_objects
-vpi_control
-vpi_flush
-vpi_fopen
-vpi_free_object
-vpi_get
-vpi_get_delays
-vpi_get_file
-vpi_get_str
-vpi_get_systf_info
-vpi_get_time
-vpi_get_userdata
-vpi_get_value
-vpi_get_vlog_info
-vpi_handle
-vpi_handle_by_index
-vpi_handle_by_name
-vpi_iterate
-vpi_mcd_close
-vpi_mcd_flush
-vpi_mcd_name
-vpi_mcd_open
-vpi_mcd_printf
-vpi_mcd_vprintf
-vpi_printf
-vpi_put_delays
-vpi_put_userdata
-vpi_put_value
-vpi_register_cb
-vpi_register_systf
-vpi_remove_cb
-vpi_scan
-vpi_sim_control
-vpi_sim_vcontrol
-vpi_vprintf
-vpip_count_drivers
-vpip_format_strength
-vpip_make_systf_system_defined
-vpip_mcd_rawwrite
-vpip_set_return_value
-acc_fetch_fullname
-acc_fetch_fulltype
-acc_fetch_name
-acc_fetch_type
-acc_fetch_type_str
-mti_AddLoadDoneCB
-mti_AddQuitCB
-mti_Break
-mti_CreateProcess
-mti_CreateProcessWithPriority
-mti_Delta
-mti_Desensitize
-mti_FindRegion
-mti_FindSignal
-mti_FindVar
-mti_FirstLowerRegion
-mti_FirstLowerRegion
-mti_FirstSignal
-mti_FirstVarByRegion
-mti_GetArrayElementType
-mti_GetArraySignalValue
-mti_GetArrayVarValue
-mti_GetEnumValues
-mti_GetNumRecordElements
-mti_GetPrimaryName
-mti_GetProductVersion
-mti_GetRegionFullName
-mti_GetRegionName
-mti_GetRegionSourceName
-mti_GetResolutionLimit
-mti_GetSignalName
-mti_GetSignalNameIndirect
-mti_GetSignalSubelements
-mti_GetSignalType
-mti_GetSignalValue
-mti_GetSignalValueIndirect
-mti_GetTopRegion
-mti_GetTypeKind
-mti_GetVarKind
-mti_GetVarName
-mti_GetVarSubelements
-mti_GetVarType
-mti_GetVarValue
-mti_GetVarValueIndirect
-mti_NextRegion
-mti_NextSignal
-mti_NextVar
-mti_Now
-mti_NowUpper
-mti_Quit
-mti_ScheduleWakeup
-mti_ScheduleWakeup64
-mti_Sensitize
-mti_SetSignalValue
-mti_SetVarValue
-mti_TickLeft
-mti_TickLength
-mti_TickRight
-mti_VsimFree
-mti_Interp
-mti_Cmd
-Tcl_GetStringResult
-Tcl_ResetResult
-Tcl_GetObjResult
-Tcl_ResetResult
-Tcl_ListObjGetElements
-Tcl_GetStringResult
-TclFreeObj
-Tcl_ResetResult
-Tcl_ResetResult
-Tcl_GetString
-TclFreeObj
```

Comment:
What about all these non-vpi bits?

Reply:
These are loaded from the import library shipped with modelsim `mtipli.lib`. We already dependent on the header files shipped with modelsim so this does not introduce any additional dependency

Reply:
I will try to chek but have filling that it has to be done for fli and vhpi (tcl?) also to work without def files. 

We also remove simple def file and add more c code? Is there more advantage than this?

Reply:
Agreed, it's not clear to me what the benefit of this is.

Reply:
I have tested it, VPI and FLI for modelsim compile fine. If VHPI is also converted to runtime dynamic linking, all .def files could be dropped and the path to compile with MSVC would be unblocked. Actually, if uncommenting the aldec vhpi library and applying a bunch of hacks (adapting the compiler flags for MSVC, adding the __declspec(dllexport) where necesarry), cocotb can be installed with `pip install .` using the official binary version of python for windows.

Moreover, if the dependency on the DLL name is dropped, the `libcocotbvpi_*` could be easily merged into a single `libcocotbvpi` library.

Reply:
Additionally, icarus master, starting with [`a59b183b`](https://github.com/steveicarus/iverilog/commit/a59b183bd1890eff63cc59d38084dde15de281ff#diff-9bb3b161b30ef5d95427f8ed62cb9e88R685) is no longer working on windows because instead of exporting `vpi_*` in `vvp.exe`, now a function table is passed. As a result, the current `libcocotbvpi_icarus.vpl` is unable to load because it no longer can import the symbols. Based on this PR, `libcocotbvpi_icarus.vpl` could be easily adapted to support both versions of iverilog.

Reply:
> Additionally, icarus master, starting with [`a59b183b`](https://github.com/steveicarus/iverilog/commit/a59b183bd1890eff63cc59d38084dde15de281ff#diff-9bb3b161b30ef5d95427f8ed62cb9e88R685) is no longer working on windows because instead of exporting `vpi_*` in `vvp.exe`, now a function table is passed. As a result, the current `libcocotbvpi_icarus.vpl` is unable to load because it no longer can import the symbols. Based on this PR, `libcocotbvpi_icarus.vpl` could be easily adapted to support both versions of iverilog.

Could you create a new issue to explain and track this?

Reply:
>Additionally, icarus master, starting with a59b183b is no longer working on windows

This is a good reason to change.

>Moreover, if the dependency on the DLL name is dropped, the libcocotbvpi_* could be easily merged into a single libcocotbvpi library.

This is may hard to do for other/compatibility reasons, unfortunately.

Reply:
>I have tested it, VPI and FLI for modelsim compile fine. If VHPI is also converted to runtime dynamic linking, all .def files could be dropped and the path to compile with MSVC would be unblocked. Actually, if uncommenting the aldec vhpi library and applying a bunch of hacks (adapting the compiler flags for MSVC, adding the __declspec(dllexport) where necesarry), cocotb can be installed with pip install . using the official binary version of python for windows.

Do you know if such dll does actually work/loads without issues?

Reply:
Are def files not compatible with msvc? I thought they were.

Reply:
> Are def files not compatible with msvc? I thought they were.

It would be possible. `dlltool` is not part of the msvc toolchain and therefore would need to be substituted. Probably, it would also be necessary to locate the msvc toolchain because typically it is not on `PATH`. IMHO the extra maintenance burden of this way would object it. Whereas this PR reduces dependencies instead of introducing new ones. VPI/VHPI APIs have not really changed in years and therefore I do not expect much maintenance effort in having a wrapper for runtime loading.

Reply:
Right, the MSVC equivalent is `lib /def:modelsim.def /out:modelsim.lib /machine:x86`